services/api/service.go

```diff
@@ -835,6 +966,55 @@ func (api *RelayAPI) handleGetPayload(w http.ResponseWriter, req *http.Request)
 		if err != nil {
 			log.WithError(err).Error("failed to increment builder-stats after getPayload")
 		}
+
+		// Check if the block was valid.
+		submitBlockReq := types.BuilderSubmitBlockRequest{
+			Signature:        payload.Signature,
+			Message:          &bidTrace.BidTrace,
+			ExecutionPayload: getPayloadResp.Data,
+		}
+
+		simErr := api.simulateBlock(blockSimOptions{
+			ctx:        context.Background(),
+			log:        log,
+			isHighPrio: true, // manually set to true for these blocks.
```

Comment:
This should probably be `isHighPrio: false`. The reason is that all blocks (including optimitic ones) will get simulated within the slot and we should prioritise the simulation of high-prio-non-optimistic blocks over optimistic blocks.

Reply:
good point! updated and changed the description and diagram to reflect this :-)